internal/error.go

```diff
@@ -459,6 +477,20 @@ func NewContinueAsNewError(ctx Context, wfn interface{}, args ...interface{}) er
 	return i.NewContinueAsNewError(ctx, wfn, args...)
 }
 
+// NewContinueAsNewErrorWithOptions creates ContinueAsNewError instance with additional options.
+func NewContinueAsNewErrorWithOptions(ctx Context, options ContinueAsNewErrorOptions, wfn interface{}, args ...interface{}) error {
```

Comment:
Awkward this doesn't  go through the `NewContinueAsNewError`  interceptor, but doing it properly would require changing the interceptors type signature to take  `ContinueAsNewErrorOptions`. @cretz thoughts?

Reply:
It does go through the existing `NewContinueAsNewError` interceptor for the initial creation (which can set this value too), just not with the options, correct. I think that's ok for now.